src/parser/tptp/Tptp.g

```diff
@@ -883,7 +1034,10 @@ thfLogicFormula[CVC4::Expr& expr]
         {
           // TODO create whatever lambda
         }
-        expr = MK_EXPR(kind::EQUAL, expr, expr2);
+        args.push_back(expr);
+        args.push_back(expr2);
+        ParseOp p(kind::EQUAL);
+        expr = PARSER_STATE->applyParseOp(p, args);
```

Comment:
Nit: this is a bit over-refactored, this line could have stayed the same.  (Same applies to a few other places).

Reply:
In `applyParseOp` there are checks for using HOL stuff. We'd have to move them here otherwise.

Reply:
I see, sounds good.